examples/khr-3hv/full project/controllers/supervisorsManager/PPORun.py

```diff
@@ -0,0 +1,154 @@
+
+from khr3hvEnv import khr3hvRobotSupervisor
+
+
+# Scheduler for the learning rate when using the stable baselines
+def linear_schedule(initial_value: float) -> Callable[[float], float]:
+    """
+    Linear learning rate schedule.
+    :param initial_value:
+    :return: current learning rate depending on remaining progress
+    """
+    if isinstance(initial_value, str):
+        initial_value = float(initial_value)
+
+    # TODO  decrease the lr by .10
+    def func(progress_remaining: float) -> float:
+        """
+        Progress will decrease from 1 (beginning) to 0
+        :param progress_remaining: (float)
+        :return: (float)
+        """
+        return progress_remaining * initial_value
+
+    return func         
+            
+def run(projectName, wandbSaveFreq, train, useRay, output, args):
+
+    if useRay:
+        # Register the enviroment to be used by Ray
+        register_env("khr3hv", lambda config: khr3hvRobotSupervisor(projectName, 
+                                              wandbSaveFreq, train, useRay))
+        # Model hyperparameters
+        model_config = {
+                "env": "khr3hv",
+                "model":{
+                    "fcnet_hiddens": [256, 256],
+                    "fcnet_activation": "relu", # tanh or relu
+                },
+                "lr": args.lr,
+                "framework": "torch",
+                "gamma": args.gamma,
+                "lambda": args.gae_lambda,
+                "clip_param": args.clip_param,
+                "kl_coeff": args.kl_coeff,
+                "num_sgd_iter": args.num_sgd_iter,
+                "sgd_minibatch_size": args.sgd_minibatch_size,
+                "horizon": 200,
+                "train_batch_size": args.train_batch_size,
+                "num_workers": args.num_workers,
+                "num_gpus": args.num_gpus,
+                "batch_mode": args.batch_mode,
+                "observation_filter": args.observation_filter
+                }
+        # Stop criteria for training
+        stop = {
+                "timesteps_total": 100e6,
+        }
+        if train:
+            # Init the wandb providing the project name
+            #wandb.init(project=PROJECT_NAME)
+            # Train the model using Ray tune
+            tune.run(ppo.PPOTrainer, 
+                            config=model_config, 
+                            stop=stop,
+                            callbacks=[WandbLoggerCallback(
+                                    project=projectName,
+                                    api_key_file="../../../wandb.txt",
+                                    log_config=True)],
+                            local_dir=output, 
+                            checkpoint_freq=wandbSaveFreq,
+                            checkpoint_at_end=True
+                            )
+            ray.shutdown()
+        else:
+            # Crate an agent having the predifined model hyperparameters
```

Comment:
Add ray.init()?
When `TRAIN = False` and `USE_Ray = True`, I get _System error: Ray has not been started yet. You can start Ray with 'ray.init()'._